x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -81,6 +82,7 @@ export class JobCreator {
     }
 
     this._datafeed_config.query = query;
+    this._partitionField = null;
```

Comment:
As from our conversation, I have removed `this._partitionField` altogether. Since storing the index might lead to the same issue where the index being out of sync with the detectors (e.g. like deleting one of the detector), I have changed the getter function to find the first instance of detector where the keyword mlcategory exists. Since the `partition_field_name` property has to be the same value in every detector that uses the keyword mlcategory, I think this change is more suitable.